File: benchmark/php/cumulative_sum.php
<?php
$arr = [];
for ($i = 0; $i < 100000; $i++) { $arr[] = $i + 1; }
$start = hrtime(true);
$sum = 0;
for ($i = 0; $i < 100000; $i++) { $sum += $arr[$i]; $arr[$i] = $sum; }
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Last: " . $arr[99999] . "
";
echo "Time: " . round($ms) . "ms
";

File: benchmark/src/perl/cumulative_sum.pl
use Time::HiRes qw(time);
my @arr = 1..100000;
my $start = time();
my @cum; my $sum = 0;
for my $i (0..99999) { $sum += $arr[$i]; push @cum, $sum; }
my $end = time();
my $ms = ($end - $start) * 1000;
print "Result: $cum[99999]\n";
printf "Time: %.0fms\n", $ms;
